src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -624,24 +639,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
     End Class
 
-    Friend Partial Class BoundUserDefinedUnaryOperator
+    Partial Friend Class BoundUserDefinedUnaryOperator
         Implements IUnaryOperatorExpression
 
         Private ReadOnly Property IHasOperatorMethodExpression_OperatorMethod As IMethodSymbol Implements IHasOperatorMethodExpression.OperatorMethod
             Get
-                Return Me.Call.Method
+                Return If(Me.UnderlyingExpression.Kind = BoundKind.Call, Me.Call.Method, Nothing)
             End Get
         End Property
 
         Private ReadOnly Property IHasOperatorMethodExpression_UsesOperatorMethod As Boolean Implements IHasOperatorMethodExpression.UsesOperatorMethod
             Get
-                Return True
+                Return IHasOperatorMethodExpression_OperatorMethod IsNot Nothing
             End Get
         End Property
 
         Private ReadOnly Property IUnaryOperatorExpression_Operand As IOperation Implements IUnaryOperatorExpression.Operand
             Get
-                Return Me.Operand
+                If Me.UnderlyingExpression.Kind = BoundKind.Call Then
+                    Return Me.Operand
+                Else
+                    Return GetChildOfBadExpression(Me.UnderlyingExpression, 0)
```

Comment:
It doesn't look like we have an existing compiler unit-test hitting that specific error code path, but this test hits it, I believe:
```
        <Fact>
        Public Sub Test()
            Dim compilationDef =
<compilation>
    <file name="a.vb">
Class C1
    Public Shared Operator -(x As C1) As C1
        Return Nothing
    End Operator

    Public Shared Operator -(x As C1) As C1
        Return Nothing
    End Operator

    Sub Test()
        Dim x = -Me
    End Sub
End Class
    </file>
</compilation>

            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(compilationDef, options:=TestOptions.ReleaseExe)
            compilation.GetSemanticModel(compilation.SyntaxTrees.Single()).GetDiagnostics().AssertTheseDiagnostics(<![CDATA[
BC30269: 'Public Shared Operator -(x As C1) As C1' has multiple definitions with identical signatures.
    Public Shared Operator -(x As C1) As C1
                           ~
BC30521: Overload resolution failed because no accessible '-' is most specific for these arguments:
    'Public Shared Operator -(x As C1) As C1': Not most specific.
    'Public Shared Operator -(x As C1) As C1': Not most specific.
        Dim x = -Me
                ~~~
]]>)
        End Sub
```